compiler/src/dotty/tools/dotc/interpreter/Interpreter.scala

```diff
@@ -117,11 +117,15 @@ class Interpreter(implicit ctx: Context) {
       case Typed(expr, _) =>
         interpretTreeImpl(expr, env)
 
-      // Getting the underlying value of a value class. The value class is evaluated as its boxed representation
-      // as values in the interpreter are `Object`s. Therefore we just get it from the enviroment as is.
-      case Select(qualifier, _)
+      case Select(qualifier, name)
           if tree.symbol.owner.isValueClass && tree.symbol.is(ParamAccessor) && env.contains(qualifier.symbol) =>
-        env(qualifier.symbol)
+        val value = env(qualifier.symbol)
+        val clazz = value.getClass
+        if (clazz.getCanonicalName != tree.symbol.owner.showFullName) value // Already unboxed
```

Comment:
If the value class is a nested inner class, I guess this check will fail.

Reply:
I will add a test case and check it

Reply:
There is an issue with inner classes but it is not in this code. I will fix it another PR.